app/app/signin/page.jsx

```diff
@@ -0,0 +1,326 @@
+"use client";
+import {
+  Container,
+  TextField,
+  Button,
+  Typography,
+  Grid2,
+  Box,
+} from "@mui/material";
+import { useRouter } from "next/navigation";
+
+import { useState } from "react";
+
+const SignUp = () => {
+  const [data, setData] = useState({
+    firstName: "",
+    lastName: "",
+    email: "",
+    username: "",
+    password: "",
+  });
+  const handleChange = (e) => {
+    setErrors({
+      firstName: "",
+      lastName: "",
+      email: "",
+      username: "",
+      password: "",
+    });
+    setData((prevData) => ({ ...prevData, [e.target.name]: e.target.value }));
+  };
+
+  const [errors, setErrors] = useState({
+    firstName: "",
+    lastName: "",
+    email: "",
+    username: "",
+    password: "",
+  });
+
+  const isValidate = () => {
+    let isValid = true;
+    let tempErrors = {};
+    if (!data.firstName) {
+      tempErrors.firstName = "Firstname is required";
+      isValid = false;
+    }
+    if (!data.lastName) {
+      tempErrors.lastName = "Lastname is required";
+      isValid = false;
+    }
+    if (!data.email) {
+      tempErrors.email = "Email is required";
+      isValid = false;
+    } else if (!/\S+@\S+\.\S+/.test(data.email)) {
+      tempErrors.email = "Email is not valid";
+      isValid = false;
+    }
+    if (!data.username) {
+      tempErrors.username = "Username is required";
+    }
+    if (!data.password) {
+      tempErrors.password = "Password";
+      isValid = false;
+    }
+    setErrors(tempErrors);
+    return isValid;
+  };
+
+  const router = useRouter();
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+
+    if (isValidate()) {
+      const userData = {
+        first_name: data.firstName,
+        last_name: data.lastName,
+        email: data.email,
+        username: data.username,
+        password: data.password,
+      };
+
+      try {
+        const response = await fetch("http://localhost:3001/auth/register", {
```

Comment:
I would advise to maybe extract the response making to a separate helper function and in the components themselves you can only use for example `await makeRequest('/auth/register')` so that you extract all the reusable logic to another place

Reply:
Thanks for you time. I updated the code by making separate function